File: 0330-patching-array/0330-patching-array.py
class Solution:
    def minPatches(self, nums: List[int], n: int) -> int:
        num, idx, result = 1, 0, 0

        while num <= n:
            if idx < len(nums) and nums[idx] <= num:
                num += nums[idx]
                idx += 1
            else:
                num += num
                result += 1

        return result

File: 0330-patching-array/0330-patching-array.java
class Solution {
    public int minPatches(int[] nums, int n) {
        long num = 1;
        int idx = 0, result = 0;

        while (num <= n) {
            if (idx < nums.length && nums[idx] <= num) {
                num += nums[idx++];
            } else {
                num += num;
                result++;
            }
        }

        return result;
    }
}